thinc/layers/maxout.py

```diff
@@ -18,7 +18,7 @@
 def Maxout(
     nO: Optional[int] = None,
     nI: Optional[int] = None,
-    nP: int = 3,
+    nP: Optional[int] = 3,
```

Comment:
This doesn't need to be `Optional`, because it should always be an int I think? I don't think we need to set it to `None`.

Reply:
Yea, so we need this if we want to keep the `set_dim` error. Otherwise, during serialization, `nP` gets first set to the default value 3 and then overwritten to its actual value from disk, which throws an error. That's why now it reads `nP=None` in `test_serialize`, that test crashes otherwise.